tests/ignite/distributed/test_auto.py

```diff
@@ -43,6 +49,22 @@ def test_auto_dataloader_warning(distributed_context_single_node_gloo):
         )
 
 
+@pytest.mark.distributed
+@pytest.mark.skipif(not idist.has_native_dist_support, reason="Skip if no native dist support")
+@pytest.mark.skipif("WORLD_SIZE" not in os.environ, reason="Skip if WORLD_SIZE not in env vars")
+def test_auto_dataloader_warning_distributed_sampler(distributed_context_single_node_gloo):
+    dataset = DummyDS()
+    rank = idist.get_rank()
+    world_size = idist.get_world_size()
+    auto_dataloader(dataset, sampler=DistributedSampler(dataset, num_replicas=world_size, rank=rank))
+    expected_warning = f"Found distributed sampler with rank={rank + 1}, but process rank is {rank}"
+    with pytest.warns(UserWarning, match=expected_warning):
+        auto_dataloader(dataset, sampler=DistributedSampler(dataset, num_replicas=world_size, rank=rank + 1))
```

Comment:
Unfortunately, this does not work. Maybe, like that:

```suggestion
        auto_dataloader(dataset, sampler=DistributedSampler(
            dataset, num_replicas=world_size, rank=(rank + 1) % world_size)
        )
```
https://github.com/pytorch/ignite/pull/2119/checks?check_run_id=3113040697#step:12:2537

Reply:
Indeed, just seen the error message as well.